src/main/java/graphql/schema/diff/IntrospectionSchemaDiffSet.java

```diff
@@ -0,0 +1,88 @@
+package graphql.schema.diff;
+
+import graphql.Assert;
+import graphql.ExecutionResult;
+import graphql.GraphQL;
+import graphql.PublicApi;
+import graphql.introspection.IntrospectionQuery;
+import graphql.introspection.IntrospectionResultToSchema;
+import graphql.language.Document;
+import graphql.schema.GraphQLSchema;
+
+import java.util.Map;
+
+/**
+ * Represents 2 schemas that can be diffed.  The {@link SchemaDiff} code
+ * assumes that that schemas to be diffed are the result of a
+ * {@link graphql.introspection.IntrospectionQuery}.
+ */
+@PublicApi
+public class IntrospectionSchemaDiffSet implements SchemaDiffSet {
+
+    private final Map<String, Object> introspectionOld;
+    private final Map<String, Object> introspectionNew;
+
+    public IntrospectionSchemaDiffSet(Map<String, Object> introspectionOld, Map<String, Object> introspectionNew) {
+        this.introspectionOld = introspectionOld;
+        this.introspectionNew = introspectionNew;
+    }
```

Comment:
again as discussed below - store this as a Document not a `Map<String, Object>` (I know the old code did but we get a chance to be better with new code)

and make this constructor private